core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1601,7 +1601,7 @@ class UnifiedLog(@volatile var logStartOffset: Long,
       shouldDelete
     }
 
-    deleteOldSegments(shouldDelete, RetentionSizeBreach(this, remoteLogEnabled()))
+    deleteOldSegments(shouldDelete, RetentionSizeBreach(this, remoteLogEnabledAndRemoteCopyEnabled()))
   }
 
   private def deleteLogStartOffsetBreachedSegments(): Int = {
```

Comment:
please apply `remoteLogEnabledAndRemoteCopyEnabled` to `deleteLogStartOffsetBreachedSegments` also

Reply:
I'm not sure if this is ok as deleteLogStartOffsetBreachedSegments actually use remoteLogEnabled instead of remoteLogEnabledAndRemoteCopyEnabled

Reply:
If remote copy is disabled, we should not delete segments according to local LSO. @showuon @brandboat WDYT?

Reply:
We don't need to change `deleteLogStartOffsetBreachedSegments`. 

The delete log start offset breached will happen in cases like user sends RPC to delete old records. So the log start offset will be incremented. (ex: delete records to offset 10, so we increment log start offset to 10). And in `deleteLogStartOffsetBreachedSegments`, we'll check if the incremented log start offset need to delete segments, ex: segment 1 contains offset 0 ~ 8, so segment 1 can be deleted... etc. 

If remote storage is enabled (no matter remote copy enable or not), we will always check local log start offset, because there will be another thread in remote log manager to check the log start offset breach or not.